src/test/kotlin/no/nav/syfo/dialogmelding/SendDialogmeldingArenaCronjobSpek.kt

```diff
@@ -0,0 +1,225 @@
+import org.spekframework.spek2.style.specification.describe
+import java.sql.Timestamp
+import java.time.LocalDateTime
+import java.util.UUID
+
+class SendDialogmeldingArenaCronjobSpek : Spek({
+    describe(SendDialogmeldingArenaCronjob::class.java.simpleName) {
+        with(TestApplicationEngine()) {
+            start()
+
+            val externalMockEnvironment = ExternalMockEnvironment.instance
+            val database = externalMockEnvironment.database
+            val mqSender = mockk<MQSenderInterface>(relaxed = true)
+            val emottakService = mockk<EmottakService>(relaxed = true)
+            val smtssClient = mockk<SmtssClient>(relaxed = true)
+
+            val arenaDialogmeldingService = ArenaDialogmeldingService(
+                mqSender = mqSender,
+                smtssClient = smtssClient,
+                emottakService = emottakService,
+            )
+            val sendDialogmeldingArenaCronjob = SendDialogmeldingArenaCronjob(
+                database = database,
+                arenaDialogmeldingService = arenaDialogmeldingService,
+            )
+
+            describe("Send dialogmeldinger to Arena via cronjob") {
+                beforeEachTest {
+                    database.dropData()
+                    clearAllMocks()
+                    justRun { mqSender.sendArena(any()) }
+                    coEvery { smtssClient.findBestTss(any(), any(), any()) } returns TssId("123")
+                    coJustRun { emottakService.registerEmottakSubscription(any(), any(), any(), any(), any()) }
+                }
+
+                val fellesformat =
+                    getFileAsString("src/test/resources/dialogmelding_dialog_notat.xml")
+                val fellesformatXml = safeUnmarshal(fellesformat)
+                val receivedDialogmelding = ReceivedDialogmelding.create(
+                    dialogmeldingId = UUID.randomUUID().toString(),
+                    fellesformat = fellesformatXml,
+                    inputMessageText = fellesformat,
+                )
+
+                fun createDialogmeldingOpplysning(
+                    receivedDialogmelding: ReceivedDialogmelding,
+                    apprecStatus: Status = Status.OK
+                ): String {
+                    database.connection.use {
+                        val id = it.opprettDialogmeldingOpplysninger(
+                            receivedDialogmelding = receivedDialogmelding,
+                        )
+                        it.opprettBehandlingsutfall(
+                            validationResult = ValidationResult(
+                                status = apprecStatus,
+                                ruleHits = emptyList(),
+                            ),
+                            dialogmeldingid = id,
+                        )
+                        it.commit()
+                        return id
+                    }
+                }
+
+                it("Sends dialogmelding to arena") {
```

Comment:
Kan legge til betingelsene for at en melding skal sendes til arena i testnavnet her

Reply:
Gjorde et forsøk nå!